src/Sparrow/Json/Parsing/ObjectJsonParser.cs

```diff
@@ -577,24 +577,30 @@ private void SetStringBuffer(string str)
         {
             // max possible size - we avoid using GetByteCount because profiling showed it to take 2% of runtime
             // the buffer might be a bit longer, but we'll reuse it, and it is better than the computing cost
-            int byteCount = Encodings.Utf8.GetMaxByteCount(str.Length);
+           
             int escapePositionsSize = JsonParserState.FindEscapePositionsMaxSize(str, out _);
 
-            // If we do not have a buffer or the buffer is too small, return the memory and get more.
-            var size = byteCount + escapePositionsSize;
-            if (_currentStateBuffer == null || _currentStateBuffer.SizeInBytes < size)
+            int byteCount = str.Length * 5 + escapePositionsSize;
+            if (_currentStateBuffer == null || _currentStateBuffer.SizeInBytes < byteCount)
             {
-                if (_currentStateBuffer != null)
-                    _ctx.ReturnMemory(_currentStateBuffer);
-                _currentStateBuffer = _ctx.GetMemory(size);
-                Debug.Assert(_currentStateBuffer != null && _currentStateBuffer.Address != null);
+                byteCount = Encodings.Utf8.GetMaxByteCount(str.Length);
```

Comment:
why is this line needed?

Reply:
Because we are playing safe with 5.2. Will probably remove it entirely on 6.0 but not for now.